feat(queue): add queue freeze API tool (#4160)
Mergifyio/mergify-engine: docs/source/actions/queue.rst

draft pull requests of 5 pull requests each being tested at the same time. If
your CI time is 10 min, you can merge those 15 pull requests in only 10 minutes.

Queue Freeze
~~~~~~~~~~~~

|premium plan tag|

Mergify allows to freeze the merge of one or several queues at the same time. In order to provide maximum control
and flexibility on how and when you want the code to be merged.

When a queue is frozen, all the checks and CI configured on the queue rules are still running, it is simply and
uniquely the merge that is blocked. This allows you to continue your work, development and testing while having
the security that nothing will be deployed.

> The freeze tool is provided through the Mergify API, where you can see the references at: `API references <https://docs.mergify.com/api#/operations/create_queue_freeze_repos__owner___repository__queue__queue_name__freeze_put>`_

Author: I wasn't sure about the generic link at first, but indeed it is risky to put directly the specific link, it is better for us to expect the user to find the different links by him/herself ! gonna change that

> Mergify allows freezing the merge of one or several queues at the same time. In order to provide maximum control
> and flexibility on how and when you want the code to be merged.
> 
> When a queue is frozen, all the checks and CI configured on the queue rules are still scheduled and triggered;
> only the merge is blocked. This allows you to continue your work, development, and testing while having
> the security that no pull request will be merged from the queue.
> 
> The freeze feature is available through `the Mergify API <https://docs.mergify.com/api>`_.
> 
> A set of API endpoints allows you to create, get, update and delete freezes on your queues.
> 
> For example, let's say you have three queues, respectively ``hotfix``, ``default`` and ``lowprio``, in this order.
> 
> You are in the middle of an incident for your project, and want your production branch
> to not receive anything else than bug fixes. By creating a freeze on the ``default`` queue, you are now
> assured that nothing from this queue  — nor the queue after it, ``lowprio`` — will be merged.
> 
> Meanwhile, you can still merge pull requests that are sent to the ``hotfix`` queue.
> 
> Once your incident is resolved, you can unfreeze the queue ``default`` and returns to the previous state.

Reviewer: ```suggestion
Mergify allows freezing the merge of one or several queues simultaneously to provide maximum control
and flexibility on how and when you want the code to be merged.

The freeze feature is available through `the Mergify API <https://docs.mergify.com/api>`_. A set of API endpoints allows you to create, get, update and delete freezes on your queues.

When a queue is frozen, all the checks and CI configured on the queue rules are still scheduled and triggered;
only the merge is blocked. Freezing allows you to continue your work, development, and testing while ensuring that no pull requests will be merged from the queue.

For example, let's say you have three queues, respectively ``hotfix``, ``default`` and ``lowprio``, in this order.

You are in the middle of an incident for your project and want your production branch
not to receive anything other than bug fixes. By creating a freeze on the ``default`` queue, you are now
assured that nothing from this queue  — nor the queues after it, ``lowprio`` — will be merged.

Meanwhile, you can still merge pull requests pushed to the ``hotfix`` queue.

Once your incident is resolved, you can unfreeze the queue ``default`` and returns to the previous state.
```

> 

Author: Yes, I think with time and the API growing we might find a solution for better use case explanations !
